### ISSUE TITLE
A Merge() operation loses the database Flock.

### ISSUE THREAD

**Issue by prologic** [prologic/bitcask#200]
@yashschandra I _think_ some recent work you did caused a regression here.

I've put up a branch that shows this failure as a test case. We _should_ probably fix it 😀

```#!sh
=== RUN   TestLockingAfterMerge
    bitcask_test.go:1648:
        	Error Trace:	bitcask_test.go:1648
        	Error:      	An error is expected but got nil.
        	Test:       	TestLockingAfterMerge
--- FAIL: TestLockingAfterMerge (0.08s)
FAIL
coverage: 89.7% of statements in github.com/prologic/bitcask
FAIL	github.com/prologic/bitcask	2.915s
FAIL
make: *** [test] Error 1
```

See `regression_locking_after_merge` branch.

**Comment by yashschandra** [comment]
Found the line which is causing this - https://github.com/prologic/bitcask/blob/master/bitcask.go#L483
When calling `close`, we unlock `Flock` instance. I remember copying this from older version, do you think it has been there all along? 😄

**Comment by yashschandra** [comment]
@prologic have a look at the fix #201

**Comment by prologic** [comment]
> Found the line which is causing this - https://github.com/prologic/bitcask/blob/master/bitcask.go#L483
> When calling `close`, we unlock `Flock` instance. I remember copying this from older version, do you think it has been there all along? 😄

Probably :D
